File: manual/data/java/tests/test_catboost_params.java
package com.example;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

public class test_catboost_params {
    private function_catboost_params paramFetcher;

    @BeforeEach
    void setUp() {
        paramFetcher = new function_catboost_params();
    }

    @Test
    void test_return_type() {
        String result = paramFetcher.get_model_params(150, 7, 0.1f);
        assertNotNull(result);
        assertTrue(result instanceof String);
    }

    @Test
    void test_check_iterations() {
        String result = paramFetcher.get_model_params(200, 6, 0.1f);
        assertTrue(result.contains("Iterations: 200"));
    }

    @Test
    void test_check_depth() {
        String result = paramFetcher.get_model_params(100, 10, 0.05f);
        assertTrue(result.contains("Depth: 10"));
    }

    @Test
    void test_check_learning_rate() {
        String result = paramFetcher.get_model_params(100, 6, 0.01f);
        assertTrue(result.contains("Learning Rate: 0.01"));
    }

    @Test
    void test_multiple_conditions() {
        String result = paramFetcher.get_model_params(250, 8, 0.15f);
        assertTrue(result.contains("Iterations: 250"));
        assertTrue(result.contains("Depth: 8"));
        assertTrue(result.contains("Learning Rate: 0.15"));
    }
}

File: manual/data/python/tests/test_catboost_params.py
import unittest
from function_catboost_params import CatBoostParamFetcher

class TestCatBoostParamFetcher(unittest.TestCase):
    def setUp(self):
        self.param_fetcher = CatBoostParamFetcher()

    def test_return_type(self):
        result = self.param_fetcher.get_model_params(150, 7, 0.1)
        self.assertIsInstance(result, str)

    def test_check_iterations(self):
        result = self.param_fetcher.get_model_params(200, 6, 0.1)
        self.assertIn("Iterations: 200", result)

    def test_check_depth(self):
        result = self.param_fetcher.get_model_params(100, 10, 0.05)
        self.assertIn("Depth: 10", result)

    def test_check_learning_rate(self):
        result = self.param_fetcher.get_model_params(100, 6, 0.01)
        self.assertIn("Learning Rate: 0.01", result)

    def test_multiple_conditions(self):
        result = self.param_fetcher.get_model_params(250, 8, 0.15)
        self.assertIn("Iterations: 250", result)
        self.assertIn("Depth: 8", result)
        self.assertIn("Learning Rate: 0.15", result)

if __name__ == "__main__":
    unittest.main()
